src/Compilers/CSharp/Portable/FlowAnalysis/VariableUsePass.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// VariableUsePass marks all referenced variables in the visited node as used (as a read).
+    /// This pass is used for things like attribute arguments and default parameter expressions,
+    /// which are otherwise not analyzed for use references.
+    /// </summary>
+    internal class VariableUsePass : BoundTreeWalkerWithStackGuard
```

Comment:
>BoundTreeWalkerWithStackGuard [](start = 37, length = 29)

What is the goal for inheriting from BoundTreeWalkerWithStackGuard? I believe BoundTreeWalkerWithStackGuard is going to throw InsufficientExecutionStackException when we are about to get into a stack overflow. Unless we are catching that exception and producing a diagnostics, inheriting from BoundTreeWalkerWithStackGuard has almost no benefits.